### PR TITLE
Add automated report for how many tricks are documented

### PR DESCRIPTION
**Pull request by henriquegemignani**
image

### CODE REVIEW

**Comment by toasterparty** [comment]
Sometimes, if multiple tricks within an "or" block are documented by the same video, we put the video higher up in the logic expression tree. Is this case handled by your counting method?

**Comment by Miepee** [comment]
IMO if the same video covers multiple tricks in an OR block, then each thing should still have a comment but appropriately timestamped

Also, tests fail.

**Comment by henriquegemignani** [comment]
> Sometimes, if multiple tricks within an "or" block are documented by the same video, we put the video higher up in the logic expression tree. Is this case handled by your counting method?

It does not. I meant to open this PR a month ago but kept struggling with these cases, so I'm going with the simplest possible.

File: randovania/cli/database.py [hunk 1]
```diff
@@ -388,7 +389,7 @@ def pickups_per_area_command(sub_parsers):
     parser.set_defaults(func=pickups_per_area_command_logic)
 
 
-def _find_uncommented_tricks(requirement: Requirement):
+def _find_tricks_usage_documentation(requirement: Requirement) -> typing.Iterator[str, bool]:
```
Comment (Miepee) [inline randovania/cli/database.py:392]: mypy

File: randovania/cli/database.py [hunk 2]
```diff
@@ -397,63 +398,74 @@ def _find_uncommented_tricks(requirement: Requirement):
...
+                    for req in trick_resources
+                )
+            ),
+            requirement.comment is not None,
         )
 
 
-def uncommented_trick_usages_logic(args):
+def _flat_trick_usage(requirement: Requirement) -> dict[str, bool]:
+    doc = {}
```
Comment (Miepee) [inline randovania/cli/database.py:418]: mypy

File: randovania/cli/database.py [hunk 3]
```diff
@@ -397,63 +398,74 @@ def _find_uncommented_tricks(requirement: Requirement):
...
+    for source, connections in area.connections.items():
+        paths[source.name] = {}
+        for target, requirement in connections.items():
+            trick_documentation = _flat_trick_usage(requirement)
+            if trick_documentation:
+                paths[source.name][target.name] = trick_documentation
+    return paths
+
+
+def trick_usage_documentation_logic(args):
```
Comment (Miepee) [inline randovania/cli/database.py:435]: mypy

File: randovania/cli/database.py [hunk 4]
```diff
@@ -397,63 +398,74 @@ def _find_uncommented_tricks(requirement: Requirement):
...
+                        for target_name, trick_documentation in connections.items():
+                            lines.append(f"### {source_name} -> {target_name}:")
+                            for it, used in trick_documentation.items():
+                                lines.append(f"- [{'X' if used else ' '}] {it}")
 
     output_path.write_text("\n".join(lines))
 
 
-def uncommented_trick_usages_command(sub_parsers):
+def trick_usage_documentation_command(sub_parsers):
```
Comment (Miepee) [inline randovania/cli/database.py:457]: mypy

File: randovania/cli/database.py [hunk 5]
```diff
@@ -397,66 +400,77 @@ def _find_uncommented_tricks(requirement: Requirement):
...
+                        for target_name, trick_documentation in connections.items():
+                            lines.append(f"### {source_name} -> {target_name}:")
+                            for it, used in trick_documentation.items():
+                                lines.append(f"- [{'X' if used else ' '}] {it}")
 
     output_path.write_text("\n".join(lines))
 
 
-def uncommented_trick_usages_command(sub_parsers):
+def trick_usage_documentation_command(sub_parsers: _SubParsersAction) -> None:
```
Comment (Miepee) [inline randovania/cli/database.py:459]: Do we not have a test for this subcommand? Codecv doesn't seem to report anything at all.

Comment (henriquegemignani) [reply randovania/cli/database.py:459]: probably not, a lot of the CLI things does not have tests

Comment (Miepee) [reply randovania/cli/database.py:459]: would be nice to have tho.

Comment (henriquegemignani) [reply randovania/cli/database.py:459]: Get my very low effort test, all I can have for you today

**Review by Miepee — APPROVED** [review]

**Pull request merged**